artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -742,6 +756,25 @@ public void addConsumer(ConsumerInfo info) throws Exception {
       }
    }
 
+   public void setConnectionEntry(ConnectionEntry connectionEntry) {
+      this.connectionEntry = connectionEntry;
+   }
+
+   public void setUpTtl(final long inactivityDuration, final long inactivityDurationInitialDelay, final boolean useKeepAlive) {
+      this.useKeepAlive = useKeepAlive;
+      this.maxInactivityDuration = inactivityDuration;
+
+      protocolManager.getScheduledPool().schedule(new Runnable() {
```

Comment:
@gaohoward  I guess this fix is not valid then. If you set the server to send Pings to client faster than the client is configured you will later miss another ping on the client.

this was a TODO before and I thought you were fixing it. The server must ping at a sustained rate regardless of what the client sends.